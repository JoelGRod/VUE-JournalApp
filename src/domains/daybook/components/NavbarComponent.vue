<template>
    <nav class="navbar bg-primary">
        <a class="navbar-brand text-white">
            <img src="@/assets/logo.png" 
                alt="Vue Logo"
                height="24"
                class="d-inline-block align-text-top mx-2">
            Daybook
        </a>

        <div class="d-flex">
            <a class="navbar-brand text-white">
            {{ userName }}
            </a>
            <button 
                class="btn btn-outline-info mx-2" 
                @click="$emit('on:logout')">
                <i class="fa fa-sign-out-alt"></i>
            </button>
        </div>
    </nav>
</template>

<script>

export default {
    name: 'navbar-component',
    props: {
        userName: {
            type: String,
            required: true,
            default: 'testing'
        }
    },
    emits: ["on:logout"],
}
</script>
