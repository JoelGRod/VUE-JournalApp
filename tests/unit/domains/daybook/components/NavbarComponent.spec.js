import { shallowMount } from "@vue/test-utils";

import NavbarComponent from "@/domains/daybook/components/NavbarComponent";
import createVuexStore from "../../../mock-data/mock-store";

describe("Testing: NavbarComponent.vue", () => {
  // Not necessary but put it here for demonstrative purposes
  const store = createVuexStore({
    user: {
      name: "Testing Store",
      email: "testing@store.com",
    },
    status: "authenticated",
    idToken: "abc",
    refreshToken: "xyz",
  });

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(NavbarComponent, {
      global: {
        plugins: [store],
      },
      props: {
        userName: "testing",
      },
    });
  });

  test("should match with the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should show testing in username prop', () => {
      // Arrange
      const nameElement = wrapper.find('.d-flex .navbar-brand.text-white').text();
      const nameProp = wrapper.props('userName');
      // Assert
      expect( nameElement ).toBe( nameProp );
  });

  test('logout button should emit on:logout custom event', async () => {
      // Arrange
      const button = wrapper.find('button.btn.btn-outline-info');
      // Act
      await button.trigger('click');
      // Assert
      expect( wrapper.emitted('on:logout') ).toBeTruthy();
  });
});
