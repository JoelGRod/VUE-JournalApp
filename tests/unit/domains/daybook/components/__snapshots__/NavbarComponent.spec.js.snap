// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: NavbarComponent.vue should match with the snapshot 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white"><img src="" alt="Vue Logo" height="24" class="d-inline-block align-text-top mx-2"> Daybook </a>
  <div class="d-flex"><a class="navbar-brand text-white">testing</a><button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
